src/providers/krb5/krb5_auth.c

```diff
@@ -1138,7 +1138,9 @@ static void krb5_auth_done(struct tevent_req *subreq)
 
     if (kr->is_offline) {
         if (dp_opt_get_bool(kr->krb5_ctx->opts,
-                            KRB5_STORE_PASSWORD_IF_OFFLINE)) {
+                            KRB5_STORE_PASSWORD_IF_OFFLINE)
+                && sss_authtok_get_type(pd->authtok)
+                            == SSS_AUTHTOK_TYPE_PASSWORD) {
```

Comment:
There was added a check for `authtok` type password. Similar check is also in `krb5_auth_cache_creds` at the beginning.

```
247 static void krb5_auth_cache_creds(struct krb5_ctx *krb5_ctx,
 248                                   struct sss_domain_info *domain,
 249                                   struct confdb_ctx *cdb,
 250                                   struct pam_data *pd, uid_t uid,
 251                                   int *pam_status, int *dp_err)
 252 {
 253     const char *password = NULL;
 254     errno_t ret;
 255 
 256     if (sss_authtok_get_type(pd->authtok) != SSS_AUTHTOK_TYPE_PASSWORD) {
 257         DEBUG(SSSDBG_MINOR_FAILURE,
 258               "Delayed authentication is only available for password "
 259               "authentication (single factor).\n");
 260         return;
 261     }
```

After this change that part of code will be dead in `krb5_auth_cache_creds`. Should we remove it?
Or should we set `pam_status` and `dp_err` the same as in else branch,
Because debug message in `krb5_auth_cache_creds` is a little bit more suitable in context then message in else branch *Backend is marked offline, retry later!*